pkg/activator/net/revision_backends_test.go

```diff
@@ -389,7 +390,14 @@ func epSubset(port int32, portName string, ips []string) *corev1.EndpointSubset
 		}},
 	}
 	for _, ip := range ips {
-		ss.Addresses = append(ss.Addresses, corev1.EndpointAddress{IP: ip})
+		// Randomly put the ip address in either Addresses or NotReadyAddresses, because
```

Comment:
I think it would be nicer to be explicit about this here and then make different test cases for addresses in ready addresses, not ready, and both. One pattern that may help is to create a With(NotReady)Addresses(*corev1.Endpoint) *corev1.Endpoint function which injects addresses in to an endpoint object so test cases can easily call this around the endpoint returned here.

Reply:
Yes please, don't make the tests randomized or we'll have a bad time if there's a slight bug in the implementation somewhere.